src/main/java/com/example/pokebattlez/battle/model/BattlePokemon.java

```diff
@@ -1,21 +1,23 @@
 package com.example.pokebattlez.battle.model;
 
+import com.example.pokebattlez.battle.service.PokemonService;
+import com.example.pokebattlez.model.entity.Pokemon;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
 import java.util.*;
-import java.util.concurrent.Callable;
 import java.util.function.Function;
 
 import static com.example.pokebattlez.battle.model.Stat.*;
 
 @Data
-@NoArgsConstructor
-@AllArgsConstructor
-public class Pokemon {
+public class BattlePokemon {
```

Comment:
All right! Now it's more clear which object is which type.  
I still don't see why not a single Pokemon that has battle capabilities.

Reply:
Wanted to have different models to facilitate the migration to microservices.